### PR TITLE
Time zones

### PR DESCRIPTION
**Pull request by KorinaGraber**
dateTime and dateTimeStatic should expect a moment object with a timezone. Show the timezone on the date-time input or to the right of the static date. For dateTime, use formatters and parsers to expose the date to the input as a string but reconstruct it as a moment when a new date is selected. The selected timezone set as the timezone on the moment. This gives us the ability to allow the user to change the time zone if we want to provide that functionality at a later date.

Updated dateTime and spinner to inherit from the InputController for shared behavior. They still use their own directive definitions. It appears that we actually can require to controller for directive definitions, so I'm not sure what the issue was for previous times when it didn't seem to work.

### CODE REVIEW

**Comment by KorinaGraber** [comment]
All components outside of the card container now use component syntax.

**Comment by KorinaGraber** [comment]
Ran `all`

File: source/components/dateTimeStatic/dateTimeStatic.ts [hunk 1]
```diff
@@ -24,20 +24,21 @@ export interface IDateTimeStaticController extends IDateTimeStaticBindings {
...
 	includeTime: boolean;
 	displayValue: string;
 	displayTimeZone: boolean;
 
 	static $inject: string[] = [__date.serviceName];
 	constructor(private dateUtility: __date.IDateUtility) {
-		this.displayValue = '';
 		if (this.dateValue != null && this.dateUtility.isDate(this.dateValue)) {
-			this.displayValue = moment(this.dateValue).format(__date.defaultFormats.dateFormat);
+			this.dateValue = moment(this.dateValue);
```
Comment (SonofNun15) [inline source/components/dateTimeStatic/dateTimeStatic.ts:37]: Before this line, what are you expecting `this.dateValue` to be? Is it OK to call `moment()` on a `moment.Moment` type?

File: source/components/dateTimeStatic/dateTimeStatic.ts [hunk 2]
```diff
@@ -24,20 +24,21 @@ export interface IDateTimeStaticController extends IDateTimeStaticBindings {
...
+
+			this.displayValue = this.dateValue.format(__date.defaultFormats.dateFormat);
```
Comment (SonofNun15) [inline source/components/dateTimeStatic/dateTimeStatic.ts]: This is redundant with line 41, so perhaps it should just move into an else for perf reasons?

File: source/components/dateTimeStatic/dateTimeStatic.ts (hunk 1 above)
Comment (KorinaGraber) [inline source/components/dateTimeStatic/dateTimeStatic.ts:37]: Yes it is. moment() seems to function like $q.when() in that it's a way to verify that it's a moment.

File: source/components/dateTimeStatic/dateTimeStatic.ts (hunk 2 above)
Comment (SonofNun15) [inline source/components/dateTimeStatic/dateTimeStatic.ts]: Nevermind, looking more closely I see that 41 is composed with this assignment.

Comment (KorinaGraber) [inline source/components/dateTimeStatic/dateTimeStatic.ts]: Updated anyway. The if here will just format using the dateTimeFormat.

File: source/components/dateTime/dateTime.ts [hunk 3]
```diff
@@ -159,7 +159,16 @@ function dateTime(moment: moment.MomentStatic
 			let max: string | Date | moment.Moment
 				= dateTime.max != null ? dateTime.max : defaults.maxDate;
 
-			scope.$watch((): any => { return ngModel.$viewValue; }, (newValue: any): void => {
+			ngModel.$formatters.push((value: moment.Moment): string => {
+				return moment(value).format(getFormatOrDefault());
```
Comment (SonofNun15) [inline source/components/dateTime/dateTime.ts]: Again: interesting that you are calling `moment(value)` on an object that, by the typings, appears to already be a `moment` object... ?

**Comment by SonofNun15** [comment]
:shipit:

**Pull request merged**